### PR TITLE
Add API test for /auth/projects with empty role assignments (Issue #515)

### LINKED ISSUE #515: Add test to the /auth/projects with empty assignments list being returned
As described in the parent issue an explicit test should be added to the api implementation with an empty assignments list being returned from the provider

### PR DESCRIPTION
**Pull request by MohidSheraz**
Closes #515

Added an API test to verify that when a user has no role assignments, 
the /auth/projects endpoint returns an empty list instead of all projects.

This is a regression test for the bug described in the parent issue #513.

Test: test_auth_projects_empty_when_no_roles
- Attempts to list projects for a user without authentication/roles
- Verifies either 401 (unauthorized) or empty project list
- Should NOT return all projects in the system

### CODE REVIEW

File: tests/api/src/auth/project.rs [hunk 1]
```diff
@@ -57,3 +57,85 @@ async fn test_list_user_projects() -> Result<()> {
...
 }
+
+
+#[tokio::test]
+async fn test_auth_projects_returns_empty_for_user_without_roles() -> Result<()> {
+    // Regression test for issue #515:
+    // When a user has no role assignments, /auth/projects should return
+    // an empty list, not all projects in the system.
+    
+    let mut admin_client = TestClient::default()?;
```
Comment (gtema) [inline tests/api/src/auth/project.rs:68]: Please do not use `TestClient` and instead use to openstack_sdk AsyncOpenStack (https://github.com/openstack-experimental/keystone/blob/main/tests/api/src/role/imply/check.rs#L30C16-L30C63)

**Review by gtema — CHANGES_REQUESTED** [review]
